In [1]:
from sklearn.datasets import load_iris
iris = load_iris()

In [39]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import cross_val_score
import numpy as np

In [8]:
LR_model = LogisticRegression(max_iter = 1000)

In [4]:
SVM_model = SVC()

In [5]:
DT_model=DecisionTreeClassifier()

In [6]:
RF_model = RandomForestClassifier()

**Logistic Regression**

In [40]:
from numpy.core.fromnumeric import mean
scores=(cross_val_score(LR_model,iris.data,iris.target,cv=3))
np.average(scores)

0.9733333333333333

**Support Vector Machine (SVM)**

In [41]:
from numpy.core.fromnumeric import mean
scores=(cross_val_score(SVM_model,iris.data,iris.target,cv=3))
np.average(scores)

0.96

**Decision Tree**

In [42]:
from numpy.core.fromnumeric import mean
mean(cross_val_score(DT_model,iris.data,iris.target,cv=3))

0.9666666666666667

**Random Forest**

In [43]:
from numpy.core.fromnumeric import mean
mean(cross_val_score(RF_model,iris.data,iris.target,cv=3))

0.9666666666666667

**Best Result is from Logistic Regression Which is 0.973333**